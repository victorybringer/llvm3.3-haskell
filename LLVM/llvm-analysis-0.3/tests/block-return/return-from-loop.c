extern int g(int);
extern int z;
int f(int x) {

  while(x > 10) {
    if(g(x) < 0) return -1;

    z += x;
    return 10;
  }

  for(int i = 0; i < x; ++i) {
    z += i - x;
  }

  return -79;
}
