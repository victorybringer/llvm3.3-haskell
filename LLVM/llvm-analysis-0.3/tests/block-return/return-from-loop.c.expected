fromList [ ("%for.body", -79)
         , ("%for.cond", -79)
         , ("%for.end", -79)
         , ("%while.end", -79)
         , ("%if.end", 10)
         , ("%if.then", -1)
         ]